# Hannaford Sizing Estimate

In [16]:
#check sample file
%ls "Desktop\sample.json"

 Volume in drive C is OSDisk
 Volume Serial Number is 6A99-0C8A

 Directory of C:\Users\1447308\Desktop

12/12/2016  11:15 AM             2,266 sample.json
               1 File(s)          2,266 bytes
               0 Dir(s)  19,841,773,568 bytes free


In [24]:
#check size of 1 file
import os
print (os.stat("Desktop\sample.json").st_size)

2266


* The size for one customer is approximately **2266 Bytes**

In [45]:
#check size for 1 million customers
size_1_mill=os.stat("Desktop\sample.json").st_size*1000000
print(str(float(size_1_mill))+ " B")

2266000000.0 B


In [46]:
#convert to MB and GB
print (str(size_1_mill/float(10**6)) +" MB")
print (str(size_1_mill/float(10**9))+" GB")

2266.0 MB
2.266 GB


The size for one million customers is approximately **2.266 GB**

## Sample of one file

```json
{
  "id": "1234567890",
  "key": "1234567890",
  "value": {
    "rev": "552-ba3970cd3132ad4c295a55d4318ee56e"
  },
  "doc": {
    "_id": "1234567890",
    "_rev": "552-ba3970cd3132ad4c295a55d4318ee56e",
    "doc_type": "savings",
    "savings": {
      "Hannaford_Loyalty_Card_Num": "1234567890",
      "Coupon_Savings_Current_YTD": "123.45",
      "Coupon_Savings_Previous_YTD": "456.78",
      "Coupon_Savings_2nd_Previous_YTD": "543.21",
      "Coupon_Savings_Current_Quarter": "50.12",
      "Coupon_Savings_Previous_Quarter": "51.48",
      "Coupon_Savings_2nd_Previous_Quarter": "50.12",
      "Coupon_Savings_3rd_Previous_Quarter": "51.48",
      "Coupon_Savings_4th_Previous_Quarter": "50.12",
      "Coupon_Savings_5th_Previous_Quarter": "51.48",
      "Coupon_Savings_6th_Previous_Quarter": "50.12",
      "Coupon_Savings_7th_Previous_Quarter": "51.48",
      "Coupon_Savings_8th_Previous_Quarter": "50.12",
      "Coupon_Savings_9th_Previous_Quarter": "51.48",
      "Coupon_Savings_10th_Previous_Quarter": "50.12",
      "Coupon_Savings_11th_Previous_Quarter": "51.48",
      "Coupon_Savings_12th_Previous_Quarter": "50.12",
      "Reward_Savings_Current_YTD": "15.25",
      "Reward_Savings_Previous_YTD": "20.45",
      "Reward_Savings_2nd_Previous_YTD": "14.22",
      "Reward_Earned_Current_Quarter": "3.25",
      "Reward_Eligible_Current_Quarter": "3.35",
      "Reward_Redeemed_Current_Quarter": "2.35",
      "Reward_Savings_Previous_Quarter": "2.45",
      "Reward_Savings_2nd_Previous_Quarter": "0.00",
      "Reward_Savings_3rd_Previous_Quarter": "5.66",
      "Reward_Savings_4th_Previous_Quarter": "4.89",
      "Reward_Savings_5th_Previous_Quarter": "10.54",
      "Reward_Savings_6th_Previous_Quarter": "9.56",
      "Reward_Savings_7th_Previous_Quarter": "8.51",
      "Reward_Savings_8th_Previous_Quarter": "4.87",
      "Reward_Savings_9th_Previous_Quarter": "4.23",
      "Reward_Savings_10th_Previous_Quarter": "4.11",
      "Reward_Savings_11th_Previous_Quarter": "2.11",
      "Reward_Savings_12th_Previous_Quarter": "0.47",
      "Rewards_Forfeited_Previous Quarter": "0.00",
      "Rewards_Forfeited_YTD": "2.85",
      "last_update_timestmp": "2016-12-05 13:46:17"
    }
  }
}
```

In [ ]:
## Growth Rate Estimate = 20%
## https://techcrunch.com/2013/08/24/how-fast-should-you-be-growing/
growth_rate = 0.1